c/driver_manager/adbc_driver_manager.cc

```diff
@@ -782,11 +782,14 @@ AdbcStatusCode AdbcLoadDriverFromInitFunc(AdbcDriverInitFunc init_func, int vers
   }
 
   auto result = init_func(version, raw_driver, error);
+  if (version == ADBC_VERSION_1_1_0 && result == ADBC_STATUS_NOT_IMPLEMENTED) {
+    result = init_func(ADBC_VERSION_1_0_0, raw_driver, error);
+  }
```

Comment:
maybe create a static const list of versions and just go through them in order until we get something other than `ADBC_STATUS_NOT_IMPLEMENTED` calling `init_func`? That way future updates are just adding the version to the list?